src/modules/catalog.js

```diff
@@ -1584,6 +1584,109 @@ class Catalog {
     }
   }
 
+  /**
+   * Send patch delta catalog files to patch the current catalog
+   *
+   * @function patchCatalog
+   * @param {object} parameters - Additional parameters for catalog details
+   * @param {string} parameters.section - The section that you want to update
+   * @param {string} [parameters.notification_email] - An email address where you'd like to receive an email notifcation in case the task fails.
+   * @param {boolean} [parameters.force] - Process the catalog even if it will invalidate a large number of existing items. Defaults to False.
+   * @param {file} [parameters.items] - The CSV file with all new items
+   * @param {file} [parameters.variations] - The CSV file with all new variations
+   * @param {file} [parameters.item_groups] - The CSV file with all new item_groups
+   * @returns {Promise}
+   * @see https://docs.constructor.io/rest_api/full_catalog
+   */
+  async patchCatalog(parameters = {}) {
+    let requestUrl;
+    const queryParams = { patch_delta: true };
+    const formData = new FormData();
+    const fetch = (this.options && this.options.fetch) || nodeFetch;
+
+    if (parameters) {
```

Comment:
Next 60 lines are shared between `replaceCatalog`, `updateCatalog`, and `patchCatalog`. Would it make sense to extract this piece to a helper function in this PR on in another one?

Reply:
I'm happy to ship this as is and do the refactor in a separate PR - big fan of incremental updates 😄 

Want to throw a story on the board?

Reply:
Done!